includes/admin/class-wc-admin-setup-wizard.php

```diff
@@ -549,6 +551,15 @@ public function run_deferred_actions() {
 		$this->close_http_connection();
 		foreach ( $this->deferred_actions as $action ) {
 			call_user_func_array( $action['func'], $action['args'] );
+
+			// Clear the background installation flag if this is a plugin.
```

Comment:
You can as easily implement this safeguards for themes too. I think it's worth it, plus you wouldn't need this `if('background_installer'...)`

Reply:
I didn't choose to do this initially because we _should_ add some namespacing to the option so that a plugin and theme with the same `id` won't cause unknown state.

Reply:
Fair enough. We're only installing one theme at one point anyway.